tests/test_group.rs

```diff
@@ -17,97 +18,149 @@ fn create_commit_optional_path() {
-        &bob_identity.get_signature_key_pair().get_private_key(), // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
-        Credential::from(MLSCredentialType::Basic(bob_credential)), // TODO: this consumes the credential!
-        Vec::new(),
+        bob_signature_key,
+        Credential::from(MLSCredentialType::Basic(bob_credential)),
+        mandatory_extensions.clone(),
     );
     let bob_key_package = bob_key_package_bundle.get_key_package();
 
     let alice_update_key_package_bundle = KeyPackageBundle::new(
         ciphersuite_name,
-        &alice_identity.get_signature_key_pair().get_private_key(), // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
+        alice_signature_key,
         Credential::from(MLSCredentialType::Basic(alice_credential)),
-        Vec::new(),
+        mandatory_extensions,
     );
     let alice_update_key_package = alice_update_key_package_bundle.get_key_package();
+    assert!(alice_update_key_package.verify().is_ok());
 
     // Alice creates a group
     let group_id = [1, 2, 3, 4];
-    let group_alice_1234 = MlsGroup::new(&group_id, ciphersuite_name, alice_key_package_bundle);
+    let mut group_alice_1234 = MlsGroup::new(&group_id, ciphersuite_name, alice_key_package_bundle);
 
-    // Alice adds Bob
+    // Alice adds Bob with forced self-update
     let bob_add_proposal = group_alice_1234.create_add_proposal(
         group_aad,
-        &alice_identity.get_signature_key_pair().get_private_key(),
+        alice_signature_key,
         bob_key_package.clone(),
     );
+    let epoch_proposals = vec![bob_add_proposal];
+    let (mls_plaintext_commit, _welcome_bundle_alice_bob_option) = match group_alice_1234
+        .create_commit(group_aad, alice_signature_key, epoch_proposals, true)
+    {
+        Ok(c) => c,
+        Err(e) => panic!("Error creating commit: {:?}", e),
+    };
+    let commit = match &mls_plaintext_commit.content {
+        MLSPlaintextContentType::Commit((commit, _)) => commit,
+        _ => panic!(),
+    };
+    assert!(commit.path.is_some());
 
-    // Alice updates
-    let alice_update_proposal = group_alice_1234.create_update_proposal(
+    // Alice adds Bob without forced self-update
+    let bob_add_proposal = group_alice_1234.create_add_proposal(
         group_aad,
-        &alice_identity.get_signature_key_pair().get_private_key(),
-        alice_update_key_package.clone(),
+        alice_signature_key,
+        bob_key_package.clone(),
     );
-
-    // Only AddProposals
-    let (commit_mls_plaintext, _welcome_option) = match group_alice_1234.create_commit(
-        group_aad,
-        &alice_identity.get_signature_key_pair().get_private_key(),
-        vec![bob_add_proposal.clone()],
-        false,
-    ) {
+    let epoch_proposals = vec![bob_add_proposal];
+    let (mls_plaintext_commit, welcome_bundle_alice_bob_option) = match group_alice_1234
```

Comment:
So we add Bob twice to the same group? Shouldn't this fail?

Reply:
We only create a proposal/commit that we don't apply. This should not fail.

Reply:
Then there are comments missing explaining what this does (as mentioned in another comment already).

Reply:
🤔